src/org/zaproxy/zap/extension/ascanrulesAlpha/TestUserAgent.java

```diff
@@ -0,0 +1,145 @@
+    private static final String iPhone3 = "Mozilla/5.0 (iPhone; U; CPU iPhone OS 3_0 like Mac OS X; en-us) AppleWebKit/528.18 (KHTML, like Gecko) Version/4.0 Mobile/7A341 Safari/528.16";
+
+    private static final String[] USER_AGENTS = {
+            InternetExplorer8,
+            InternetExplorer7,
+            InternetExplorer6,
+            GoogleBot2_1,
+            MsnBot1_1,
+            YahooSlurp,
+            iPhone3
+    };
+
+    @Override
+    public int getId() {
+        return PLUGIN_ID;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString("ascanalpha.testuseragent.name");
+    }
+
+    @Override
+    public String[] getDependency() {
+        return new String[0];
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("ascanalpha.testuseragent.desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.INFO_GATHER;
+    }
+
+    @Override
+    public String getSolution() {
+        return "";
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString("ascanalpha.testuseragent.refs");
+    }
+
+    @Override
+    public void init() {
+    }
+
+    @Override
+    public void scan() {
+        for(String userAgent: USER_AGENTS)
+        {
+            attack(userAgent);
+        }
+    }
+
+    private void attack(String userAgent){
+        HttpMessage newMsg = sendUserAgent(userAgent);
+        if(newMsg != null && isResponseDifferentFromOriginal(newMsg)){
+            createAlert(newMsg, userAgent);
+        }
+    }
+
+    private HttpMessage sendUserAgent(String userAgent){
+        try {
+            HttpMessage newMsg = getNewMsg();
+            HttpRequestHeader header = newMsg.getRequestHeader();
+            header.setHeader("User-Agent", userAgent);
+            sendAndReceive(newMsg);
+            return newMsg;
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    private boolean isResponseDifferentFromOriginal(HttpMessage newMsg){
+        HttpMessage originalMsg = this.getBaseMsg();
+        return isStatusCodeDifferent(newMsg, originalMsg) || isBodyDifferent(newMsg, originalMsg);
+    }
+
+    private boolean isStatusCodeDifferent(HttpMessage newMsg, HttpMessage originalMsg) {
+        return originalMsg.getResponseHeader().getStatusCode() != newMsg.getResponseHeader().getStatusCode();
+    }
+
+    private boolean isBodyDifferent(HttpMessage newMsg, HttpMessage originalMsg) {
+        return originalMsg.getResponseBody().hashCode() != newMsg.getResponseBody().hashCode();
+    }
+
+    private void createAlert(HttpMessage newMsg, String userAgent) {
+        bingo(Alert.RISK_INFO,
+                Alert.CONFIDENCE_MEDIUM,
+                this.getBaseMsg().getRequestHeader().getURI().toString(),
+                "(Header) User-Agent: " + userAgent,
```

Comment:
The attack value (`userAgent`) should not be part of the parameter.
The string ("Header") should be i18n'ed.